spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/InfraConfiguration.java

```diff
@@ -16,23 +16,109 @@
+import org.axonframework.serialization.upcasting.event.EventUpcaster;
+import org.axonframework.spring.config.MessageHandlerLookup;
+import org.axonframework.spring.config.SpringAggregateLookup;
+import org.axonframework.spring.config.SpringAxonConfiguration;
+import org.axonframework.spring.config.SpringConfigurer;
+import org.axonframework.spring.config.SpringSagaLookup;
+import org.axonframework.spring.config.annotation.HandlerDefinitionFactoryBean;
+import org.axonframework.spring.config.annotation.SpringParameterResolverFactoryBean;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.context.annotation.Role;
 
-@ConditionalOnClass(SpringAxonAutoConfigurer.class)
+import java.util.List;
+
+@ConditionalOnClass(SpringConfigurer.class)
 @AutoConfigureAfter({
         AxonAutoConfiguration.class,
         JpaAutoConfiguration.class,
         JpaEventStoreAutoConfiguration.class,
         NoOpTransactionAutoConfiguration.class,
         TransactionAutoConfiguration.class})
-@Import(SpringAxonAutoConfigurer.ImportSelector.class)
-@AnnotationDriven
 @Configuration
 public class InfraConfiguration {
+
+    @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+    @Bean
+    public static MessageHandlerLookup messageHandlerLookup() {
+        return new MessageHandlerLookup();
+    }
+
+    @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+    @Bean
+    public static SpringAggregateLookup springAggregateLookup() {
+        return new SpringAggregateLookup();
+    }
+
+    @Role(BeanDefinition.ROLE_INFRASTRUCTURE)
+    @Bean
+    public static SpringSagaLookup springSagaLookup() {
+        return new SpringSagaLookup();
+    }
+
+    @Bean
+    public SpringAxonConfiguration configuration(Configurer configurer) {
+        return new SpringAxonConfiguration(configurer);
+    }
+
+    @Bean
+    public SpringConfigurer springAxonConfigurer(ConfigurableListableBeanFactory beanFactory,
+                                                 List<ConfigurerModule> configurerModules,
+                                                 List<ModuleConfiguration> moduleConfigurations) {
+        SpringConfigurer configurer = new SpringConfigurer(beanFactory);
+        moduleConfigurations.forEach(configurer::registerModule);
+        configurerModules.forEach(c -> c.configureModule(configurer));
+        return configurer;
+    }
+
+    @Bean
+    public QueryUpdateEmitter queryUpdateEmitter(org.axonframework.config.Configuration configuration) {
```

Comment:
Had to do some digging to grasp why this is added at all. 

So, we didn't have a bean creation method for the `QueryUpdateEmitter` anywhere.
Hence, adding makes sense.
But, does it belong in this configuration class?

Wouldn't it be better placed together with the creation of the `localSegment` `QueryBus` in the `AxonAutoConfiguration` class?